lib/ansible/modules/network/icx/icx_facts.py

```diff
@@ -0,0 +1,552 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+---
+module: icx_facts
+version_added: "2.9"
+author: "Ruckus Wireless (@Commscope)"
+short_description: Collect facts from remote Ruckus ICX 7000 series switches
+description:
+  - Collects a base set of device facts from a remote device that
+    is running ICX.  This module prepends all of the
+    base network fact keys with C(ansible_net_<fact>). The facts
+    module will always collect a base set of facts from the device
+    and can enable or disable collection of additional facts.
+notes:
+  - Tested against ICX 10.1
```

Comment:
In the notes section, you can point to the icx platform guide https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/network/user_guide/platform_icx.rst

Refer: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/junos/junos_interfaces.py#L107
Applicable for other icx modules as well.

Reply:
The reason we have not added platform_icx.rst link, is currently only ssh connection is supported and that is like default. I think Junos supports other type of connections, hence they have added this link in notes. Hope this is fine.

Reply:
I think more appropriate will be when we have ansible-icx link similar to https://www.ansible.com/integrations/networks/juniper, then add that in notes.

Reply:
Refer https://github.com/ansible/ansible/pull/59914#discussion_r313247935
Same doc guideline is followed for other modules.